### ISSUE TITLE
broken API endpoints

### ISSUE THREAD

**Issue by grapesmoker** [AlexsLemonade/refinebio#2256]
### Context

I am exploring the Refine.Bio API to try and get raw data for individual samples.

### Problem or idea

I have encountered a number of problems with the API:

1. The [samples endpoint](https://api.refine.bio/v1/samples/) produces a 500 error instead of a paginated list of samples.

2. The [computed files endpoint](https://api.refine.bio/v1/computed_files/) produces a 404 instead of a paginated list of computed files. Attempting to access a specific file (e.g. https://api.refine.bio/v1/computed_files/?id=5357478) produces a 500.

3. source_url is missing from the [search endpoint](https://api.refine.bio/v1/search/?organism=homo_sapiens) although the documentation claims that this is a required return field.

### Solution or next step

Resolve the problem so that either data or proper error codes are returned from the above endpoints.

**Comment by arielsvn** [comment]
@grapesmoker thank you for filing this issue. The 500s errors are an open bug that we have https://github.com/AlexsLemonade/refinebio/issues/1562 where the requests fail intermittently. Until that one get's solved, if you need to inspect the API responses you can pass the parameter `/?format=json` to these requests to view the JSON output. Eg: [samples endpoint](https://api.refine.bio/v1/samples/?format=json) and [computed files](https://api.refine.bio/v1/computed_files/?id=5357478&format=json)

> source_url is missing from the search endpoint although the documentation claims that this is a required return field.

Where did you see that? we might have to update the documentation. The search endpoint returns limited information about each experiment. We have another endpoint to show [detailed experiment information](https://api.refine.bio/v1/#operation/experiments_read) which does include the `source_url` field. Eg [for experiment GSE24528](https://api.refine.bio/v1/experiments/GSE24528/?format=json)

**Comment by grapesmoker** [comment]
@arielsvn Thanks for the information. I'm afraid that the computed files endpoint still produces a 404 for me, but with regard to point 3, you are correct, I must have mixed something up when I was reading the documentation and was expecting a field that was not actually supposed to be there. I will edit my initial post accordingly.

**Comment by arielsvn** [comment]
@grapesmoker can you paste the URL to the computed files that you are using? This one works for me https://api.refine.bio/v1/computed_files/?format=json

**Comment by grapesmoker** [comment]
@arielsvn Ah, I see that it does work with the `format=json` parameter. Just FYI, it does _not_ work if you omit it, but the response is not a 500 like the others but rather a 404. I've attached a screenshot. I'm not sure if it's related to the problems in #1562 or not, at least it seems to be a different manifestation, but I could see it being caused by the same underlying issues.
![Screen Shot 2020-04-20 at 11 57 46 AM](https://user-images.githubusercontent.com/1723233/80033748-40d9fc80-84bb-11ea-945b-6aaa97159f41.png)

**Comment by grapesmoker** [comment]
@kurtwheeler Right, I meant to say the data for the computed files on a per-experiment basis.

**Comment by kurtwheeler** [comment]
Awesome :+1:
